1.train_models/utils/WellSubsetSum.py

```diff
@@ -0,0 +1,154 @@
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+from PlateTrainingSplits import PlateTrainingSplits
+
+
+class WellSubsetSum:
+    """
+    Select wells for the test dataset by maximizing the number of cells in the train-validation set for a given number of wells.
+    """
+
+    def __init__(self):
+        pass
+
+    def update_test_wells(self, _welldf, _category_col, _well_col, _cell_count_col, _test_well_count):
+        """
+        Parameters
+        ----------
+        _welldf: Pandas Dataframe
+            Well data with unique wells represented as rows.
+
+        _category_col: List of Strings or String
+            The categories to represent equally in the cell population.
+
+        _well_col: String
+            Well column name.
+
+        _cell_count_col: String
+            The cell count column name.
+
+        _test_well_count: Integer
+            Number of wells to sample for testing.
+
+        Returns
+        -------
+        test_wells: List of Strings
+            Test well names.
+        """
```

Comment:
I really like this doc string. You do an excellent job describing the variables. Consider adding type hints.